public/js/review-searchBar.js

```diff
@@ -0,0 +1,125 @@
+import { CLASS_LIST } from './util/cssClasses.js';
+import { delay } from './util/util.js';
+import { URL, KEYCODE } from './util/data.js';
+
+class SearchBar {
+  constructor({ domElem, inputArea, suggestions, rollingKeywords }) {
+    this.domElem = domElem;
+    this.inputArea = inputArea;
+    this.suggestions = suggestions;
+    this.rollingKeywords = rollingKeywords;
+  }
+
+  registerEvent() {
+    const { HIDDEN } = CLASS_LIST;
+
+    this.inputArea.addEventListener('focus', () => {
+      this.suggestions.classList.remove(HIDDEN);
+      this.rollingKeywords.classList.add(HIDDEN);
+    })
```

Comment:
이벤트 핸들러 함수는 비즈니스 로직을 포함하기 때문에 자주 변경되지만 이벤트 핸들러를 등록하는 부분은 자주 변경되지 않기 때문에 선언하는 부분과 등록하는 부분의 코드를 분리하면 좋은 것 같아요. 만약 아래 예시 처럼 변경한다면 비즈니스 로직을 수정할 때 `_handleInputFocus`메서드 내부에서만 수정이 일어나서 조금 더 안정적이라고 생각해요 🙂 로직이 복잡하지 않을 때는 지금 코드도 괜찮은 것 같아요.
```suggestion
    this.inputArea.addEventListener('focus', this._handleInputFocus.bind(this));
```

Reply:
아~!!!! 단순히 이벤트 핸들러가 무슨 일을 하는지 한눈에 알아보려고 함수를 분리하는 게 아니라 그런 이유도 있었군요!!!!! 대박 너무 감사합니다ㅜㅜ

Reply:
질문 너무 많이 하는 것 같아 죄송한데.. ㅜ.ㅜ 메서드에 `_`을 붙여주는 것은 private한 메서드라는 일종의 약속 같은 걸로 알고 있는데요, 왜 이벤트 핸들러가 private 이어야 하나요 ?? 😓

Reply:
이벤트 핸들러가 반드시 private한 메서드로 작성되어야 한다기 보다, 이 메서드가 `SearchBar`클래스 내부에서만 사용되는 것이라 private메서드로 구분하는게 좋다고 생각했어요 🙂